vllm/worker/worker.py

```diff
@@ -187,6 +188,9 @@ def _prepare_inputs(
                 slot = block_number * self.block_size + block_offset
                 slot_mapping.append(slot)
 
+            # Whether to use echo for prompts.
+            use_echo.append(seq_group_metadata.sampling_params.echo)
```

Comment:
Again, change `use_echo` to `get_prompt_logprobs` can be much more straightforward